src/snowflake/cli/plugins/nativeapp/teardown_processor.py

```diff
@@ -89,9 +112,60 @@ def drop_application(self, auto_yes: bool):
                     cc.message(f"Did not drop application object {self.app_name}.")
                     return  # The user desires to keep the app, therefore exit gracefully
 
-        # 4. All validations have passed, drop object
+        # 4. Check for application objects owned by the application
+        application_objects = self.get_application_objects()
+        if len(application_objects) > 0:
+            application_objects_str = self._application_objects_to_str(
+                application_objects
+            )
+            if cascade is True:
+                cc.message(
+                    dedent(
+                        f"""\
+                            The following application objects will be deleted:
+                            {application_objects_str}
+                        """
+                    )
+                )
+            elif cascade is False:
+                cc.message(
+                    dedent(
+                        f"""\
+                            The following application objects will not be deleted:
+                            {application_objects_str}
+                        """
+                    )
+                )
+            elif interactive:
+                if interactive:
+                    cascade = typer.confirm(
+                        dedent(
+                            f"""\
+                        The following application objects are owned by this application:
+                        {application_objects_str}
+                        Do you confirm to delete these?
```

Comment:
How about 

```
The following objects are owned by application <app name>:
<list goes here>
Delete these objects along with the application? [y/n/ABORT]
```

If the user just hits enter, it's not a strong enough signal, we should abort the teardown step IMHO.

Reply:
If the user didn't respond with yes/y/no/n they will be asked again ([source](https://github.com/pallets/click/blob/main/src/click/termui.py#L230-L252))

Reply:
I've mentioned a few times now that I think abort should be an option here, given that yes and no both have consequences for the user. You seem to disagree though, so I suggest you drive the discussion with the team as a whole. It's quite possible I'm a minority.

Reply:
Added the `[y/n/ABORT]` prompt